src/Symfony/Component/Form/Extension/Core/Type/CountryType.php

```diff
@@ -12,33 +12,44 @@
 namespace Symfony\Component\Form\Extension\Core\Type;
 
 use Symfony\Component\Form\AbstractType;
-use Symfony\Component\Form\ChoiceList\ArrayChoiceList;
 use Symfony\Component\Form\ChoiceList\Loader\ChoiceLoaderInterface;
+use Symfony\Component\Form\ChoiceList\Loader\IntlCallbackChoiceLoader;
 use Symfony\Component\Intl\Intl;
+use Symfony\Component\OptionsResolver\Options;
 use Symfony\Component\OptionsResolver\OptionsResolver;
 
 class CountryType extends AbstractType implements ChoiceLoaderInterface
 {
     /**
-     * Country loaded choice list.
+     * BC layer.
      *
-     * The choices are lazy loaded and generated from the Intl component.
-     *
-     * {@link \Symfony\Component\Intl\Intl::getRegionBundle()}.
-     *
-     * @var ArrayChoiceList
+     * @var ChoiceLoaderInterface
      */
-    private $choiceList;
+    private $choiceLoader;
 
     /**
      * {@inheritdoc}
      */
     public function configureOptions(OptionsResolver $resolver)
     {
         $resolver->setDefaults(array(
-            'choice_loader' => $this,
+            'choice_loader' => function (Options $options) {
+                $choiceTranslationLocale = $options['choice_translation_locale'];
+                $choiceLoader = new IntlCallbackChoiceLoader(function () use ($choiceTranslationLocale) {
+                    return array_flip(Intl::getRegionBundle()->getCountryNames($choiceTranslationLocale));
+                });
+
+                // BC layer
+                $self = clone $this;
+                $self->choiceLoader = $choiceLoader;
+
```

Comment:
Instead of deprecating each `ChoiceLoaderInterface` methods ([trigerring a lot of messages](https://travis-ci.org/symfony/symfony/jobs/362849092)) shouldn't be enough just one here? deprecating the usage of this form type as choice loader? wdyt?

Reply:
Can't we always return the new loader here and just add deprecations to the `ChoiceLoaderInterface`'s methods without modifying them? We indeed target to remove the `ChoiceLoaderInterface` from all of these types, right?

Reply:
mmh.. I guess that was what we did in https://github.com/symfony/symfony/pull/23648/files#diff-c6198934d4f35bc102ff362d40b71285R61, but is it not a BC break? 

I mean, if we always return the new loader here then, we're ignoring any custom behavior for someone overriding `loadChoiceList()` method, no? ping @ro0NL ?

Reply:
That way you can check if a API method is overridden on the legacy loader and deprecate from there, while keep calling it.

Reply:
I agree with @ro0NL. Not sure it is worth it.

Reply:
Agree, in that case I prefer [@ro0NL's version](https://github.com/symfony/symfony/pull/23648/files#diff-c6198934d4f35bc102ff362d40b71285R61), I did the changes, thanks!